### PR TITLE
support ios device enrollment

### PR DESCRIPTION
**Pull request by anzioka**

### CODE REVIEW

**Comment by anzioka** [comment]
Requires https://github.com/openlattice/chronicle-api/pull/54

File: src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentService.java [hunk 1]
```diff
@@ -86,6 +88,29 @@ private void updateDeviceIdsCache( UUID organizationId, UUID deviceEKID, String
...
+                data.put( edmCacheManager.getPropertyTypeId( MODEL_FQN ), ImmutableSet.of( device.getModel() ) );
+                data.put( edmCacheManager.getPropertyTypeId( VERSION_FQN ), ImmutableSet.of( device.getOsVersion() ) );
+            }
+
+            if ( IOSDevice.class.isAssignableFrom( datasource.get().getClass() ) ) {
+                IOSDevice device = (IOSDevice) datasource.get();
+                data.put( edmCacheManager.getPropertyTypeId( MODEL_FQN ),
+                        ImmutableSet.of( device.getLocalizedModel() ) );
+                data.put( edmCacheManager.getPropertyTypeId( VERSION_FQN ), ImmutableSet.of( device.getVersion() ) );
+                data.put( edmCacheManager.getPropertyTypeId( NAME_FQN ), ImmutableSet.of( device.getName() ) );
```
Comment (solomontang) [inline src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentService.java:107]: Why don't android devices utilize name?

Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentService.java:107]: AndroidDevice
(https://github.com/openlattice/chronicle-api/blob/main/src/main/java/com/openlattice/chronicle/sources/AndroidDevice.java) is constructed using a subset of these static properties (https://developer.android.com/reference/android/os/Build) which does not include device name. iOS (https://developer.apple.com/documentation/uikit/uidevice) surfaces name

**Review by giuliacampana — APPROVED** [review]

File: src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentService.java [hunk 2]
```diff
@@ -86,6 +88,29 @@ private void updateDeviceIdsCache( UUID organizationId, UUID deviceEKID, String
...
+        data.put( edmCacheManager.getPropertyTypeId( STRING_ID_FQN ), ImmutableSet.of( datasourceId ) );
+
+        if ( datasource.isPresent() ) {
+            if ( AndroidDevice.class.isAssignableFrom( datasource.get().getClass() ) ) {
+                AndroidDevice device = (AndroidDevice) datasource.get();
+                data.put( edmCacheManager.getPropertyTypeId( MODEL_FQN ), ImmutableSet.of( device.getModel() ) );
+                data.put( edmCacheManager.getPropertyTypeId( VERSION_FQN ), ImmutableSet.of( device.getOsVersion() ) );
+            }
+
+            if ( IOSDevice.class.isAssignableFrom( datasource.get().getClass() ) ) {
```
Comment (UnsungHero97) [inline src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentService.java:102]: Shouldn't this be an `else if () {` ?

Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentService.java:102]: I think the check is still important (robustness and clarity)

**Review by solomontang — APPROVED** [review]

**Pull request merged**